package.json

```diff
@@ -95,6 +96,8 @@
     "cid-tool": "~0.1.0",
     "cids": "~0.5.3",
     "debug": "^3.1.0",
+    "deep-extend": "~0.6.0",
+    "electron-webrtc": "~0.3.0",
```

Comment:
"electron-webrtc" and "wrtc" seem to be excluded on purpose. Could you check the reason behind this?

Reply:
Can you give me more context on 'excluded on purpose' ?

Reply:
`wrtc` and `electron-webrtc` break libp2p 

```js
/Users/hugomrdias/code/js-ipfs-ci/node_modules/libp2p/src/index.js:145
      if (t.filter(multiaddrs).length > 0) {
            ^

TypeError: t.filter is not a function
    at _modules.transport.forEach (/Users/hugomrdias/code/js-ipfs-ci/node_modules/libp2p/src/index.js:145:13)
    at Array.forEach (<anonymous>)
    at Node.start (/Users/hugomrdias/code/js-ipfs-ci/node_modules/libp2p/src/index.js:136:29)
    at gotConfig (/Users/hugomrdias/code/js-ipfs-ci/src/core/components/libp2p.js:95:26)
    at store.get (/Users/hugomrdias/code/js-ipfs-ci/node_modules/ipfs-repo/src/config.js:45:9)
    at fs.readFile (/Users/hugomrdias/code/js-ipfs-ci/node_modules/datastore-fs/src/index.js:218:7)
    at /Users/hugomrdias/code/js-ipfs-ci/node_modules/graceful-fs/graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)

```

@jacobheun can you help with this ? i have no idea what is going on :/

Reply:
some context and maybe the fix: https://github.com/libp2p/interface-transport/issues/38#issuecomment-432063553

Reply:
@hugomrdias I'll take a look.

Reply:
I found the issue, it's with `deep-extend` https://github.com/ipfs/js-ipfs/blob/master/src/core/index.js#L48. When the instance of `libp2p-webrtc-star` is passed in via options, the methods on the class end up getting stripped off. I'll push up a commit to remove `deep-extend` in favor of `@nodeutils/defaults-deep` as it doesn't mutate.

Reply:
thank you @jacobheun 
deep-extend can do immutable if you add an empty object as the first argument. Still https://www.npmjs.com/package/assign-deep is the best package for this type of stuff its one of the smallest, handles non-plain-object like promise correctly, works exactly like object.assign but deep, etc. `@nodeutils/defaults-deep` depends on full lodash so its the biggest of them all.
Please if you agree with the above pls consider using `assign-deep` everything one of my OKR is to reduce the bundle size so this would help :)

Check https://codesandbox.io/s/742p0rjvy0 for comparison

Reply:
@hugomrdias I think we can look at doing that in a subsequent PR. `deep-extend` doesn't work because it's doing cloning and turning instances into plain objects, and the filter methods get missed in the process.  We can look at `assign-deep` or one of it's sister modules, we just need to make sure it's merging objects similarly to `@nodeutils/defaults-deep`.

Reply:
agreed !